marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ZipSplitter.java

```diff
@@ -325,11 +380,25 @@ public boolean tryAdvance(Consumer<? super DocumentWriteOperation> action) {
                     return false;
                 }
 
+                splitter.count++;
                 BytesHandle nextBytesHandle = readEntry(nextEntry);
                 String name = nextEntry.getZipEntry().getName();
-
                 String uri = name;
-                if (splitter.uriTransformer != null) {
+
+                if (splitter.getUriTransformer() == null && splitter.getUriMaker() == null) {
+                    ZipSplitter.UriMakerImpl uriMaker = new ZipSplitter.UriMakerImpl();
+                    uriMaker.setInputName(splitter.inputName);
+                    splitter.setUriMaker(uriMaker);
+                }
+
+                if (splitter.getUriMaker() != null) {
+                    if (splitter.inputName != null) {
+                        splitter.getUriMaker().setInputName(splitter.inputName);
+                    }
+                    uri = splitter.getUriMaker().makeUri(splitter.count, name, nextBytesHandle);
+                }
+
+                if (splitter.getUriTransformer() != null) {
```

Comment:
For greater clarity, this could probably be an else if so execution skips this branch if getUriMaker() is not null